File: WebApi/Models/Property/Recovery.cs
namespace WebApi.Models.Property
{
    public class Recovery
    {
        public int RecoveryID { get; set; } 
        public int PropertyID { get; set; }
        public int RecoveryTypeID { get; set; }
        public string RecoveryDescription { get; set; }
        public decimal RecoveryAmount { get; set; }
        public RecoveryType RecoveryType { get; set; }
    }

    public class RecoveryType
    {
        public int RecoveryTypeID { get; set; }
        public string RecoveryTypeDescription { get; set; }
    }

    public class RecoveryRequest
    {
        public int PropertyID { get; set; }
        public int RecoveryTypeID { get; set; }
        public string RecoveryDescription { get; set; }
        public decimal RecoveryAmount { get; set; }
    }
}

File: tenant-app/src/app/shared/Property/Recovery.ts
export class Recovery {
  recoveryID: number = 0;
  propertyID: number = 0;
  recoveryTypeID: number = 0;
  recoveryDescription: string = "";
  recoveryAmount: number = 0;
  recoveryType?: RecoveryType = new RecoveryType();

  constructor(){
  }
}

export class RecoveryType {
  recoveryTypeID: number = 0;
  recoveryTypeDescription : string = "";

  constructor(){
  }
}

export class RecoveryRequest {
  propertyID: number = 0;
  recoveryTypeID: number = 0;
  recoveryDescription: string = "";
  recoveryAmount: number = 0;

  constructor(){
  }
}
